src/snowflake/snowpark/_internal/analyzer/select_statement.py

```diff
@@ -171,6 +178,17 @@ def to_subqueryable(self) -> "Selectable":
         """Some queries can be used in a subquery. Some can't. For details, refer to class SelectSQL."""
         return self
 
+    @property
+    def api_calls(self) -> Dict[str, Any]:
+        self._api_calls = self._api_calls if self._api_calls is not None else []
+        return self._api_calls
+
+    @api_calls.setter
+    def api_calls(self, value: Dict[str, Any]) -> None:
+        self._api_calls = value
+        if self._snowflake_plan:
+            self._snowflake_plan.api_calls = value
```

Comment:
Again we should be copying this list, especially when we set it in `self._snowflake_plan`

Reply:
This place doesn't need a copy because `_snowflake_plan` and the `self` (A Selectable) are different representations of the same level in the query plan.